### PR TITLE
Auto-accept completed campaign mission before exiting game from debriefing screen

### PR DESCRIPTION
**Pull request by jg18**
If the player exits the game from the options menu while at the debriefing screen after completing a mission during a campaign, then auto-accept the mission outcome.

Has not yet been scrutinized/tested.

Should fix Issue #4309.

### CODE REVIEW

**Review by Goober5000 — COMMENTED** [review]
Looks good to me.  Will hold off on approving until it is tested and out of draft status.

**Comment by jg18** [comment]
With the latest change, a manual test now passes.

File: code/menuui/optionsmenu.cpp [hunk 1]
```diff
@@ -707,8 +708,13 @@ void options_button_pressed(int n)
 		case ABORT_GAME_BUTTON:
 			gamesnd_play_iface(InterfaceSounds::USER_SELECT);
 			choice = popup( PF_NO_NETWORKING | PF_BODY_BIG, 2, POPUP_NO, POPUP_YES, XSTR( "Exit Game?", 374));
-			if ( choice == 1 )
+			if ( choice == 1 ) {
+				if (gameseq_get_state(1) == GS_STATE_DEBRIEF && (Game_mode & GM_CAMPAIGN_MODE)) {
```
Comment (jg18) [inline code/menuui/optionsmenu.cpp:712]: The only way this call could be problematic is if there are no previous game states, which isn't possible in this case.

**Review by Goober5000 — APPROVED** [review]
Good catch. 👍  Approved.

**Pull request merged**